File: Javascript/Daily Coding Problem#4.js
const findMissingPositive = (a) => {
    for (let i = 0; i < a.length; i++) {
        while (a[i] >= 1 && a[i] <= a.length && (i + 1) != a[i]) {
            let temp = a[a[i] - 1];
            a[a[i] - 1] = a[i];
            a[i] = temp;
        }
    }
    for (let i = 0; i < a.length; i++) {
        if (a[i] !== i + 1) {
            return i + 1;
        }
    }
}

console.log(findMissingPositive([1, 2, 0]));

File: C++/Daily Coding Problem#4.cpp
#include<bits/stdc++.h>
using namespace std;

int main(){
    int size;
    cout<<"Enter size of an array";
    cin>>size;
    int a[size];
    cout<<"Enter array elements";
    for(int i =0;i<size;i++){
        cin>>a[i];
    }
    for(int i =0;i<size;i++){
        while(a[i]>= 1 && a[i]<=size && (i+1) != a[i]){
            int temp = a[a[i] - 1];
            a[a[i] - 1] = a[i];
            a[i] = temp; 
        }
    }
    for(int i =0;i<size;i++){
        if(a[i] != i+1){
            cout<<i+1;
            break;
        }
    }
}